content/english/phaser-space-invaders-game/14.Activity7-player-bullet.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Activity 7"
+date: 2020-08-20T11:45:38-07:00
+draft: false
+weight: 14
+---
+
+# Allowing the Player to Shoot
+
+Now, we we want to add one of the most important parts of the game - shooting the enemies! Although we can defeat enemies by crashing into them, this won't be a good idea once we add lives later. We will create a shooting effect by creating an object for the bullet using a separate JavaScript file (like we did for the explosion).
+
+We will start with some basic stuff - loading a spritesheet and animation in Scene 1 for the player's bullets:
+
+![scene1_preload](../media/8/scene1_preload.png)
+
+The location of the spritesheet is `assets/playerBullets.png` and the frame width and height is 32:
+
+![scene1_create](../media/8/animation_with_enemy.png)
+
+Now in Scene 2, we will do something similar to what we did with the enemies and create a group for the projectiles so that we can easily manipulate them. Create a group and name it "projectiles":
+
+![projectiles](../media/8/projectiles_step1.png)
+
+Now go to PlayerBullet.js：
+
+![constructor_xy](../media/8/constructor_xy.png)
+
+Notice that, like `Explosion.js`, it extends Phaser.GameObjects.Sprite and uses the `super()` keyword in its constructor. Complete the code in the constructor. This will require that you play the bullet's animation, enable the body, and set the bullet's y-velocity to whatever speed you want your bullet to travel.
```

Comment:
```suggestion
Notice that, like `Explosion.js`, it extends `Phaser.GameObjects.Sprite` and uses the `super()` keyword in its constructor. Complete the code in the constructor. This will require that you play the bullet's animation, enable the body, and set the bullet's y-velocity to whatever speed you want your bullet to travel.
```